zerver/lib/subscription_info.py

```diff
@@ -811,21 +814,60 @@ def get_stream_id(sub_dict: RawSubscriptionDict) -> int:
             subscribed_stream_ids,
         )
 
-        # Eventually "partial subscribers" will return (at minimum):
-        # (1) all subscriptions for recently active users (for the buddy list)
-        # (2) subscriptions for all bots
-        #
-        # For now, we're only doing (2).
-        send_partial_subscribers = include_subscribers == "partial"
+        # `partial_subscribers` returns:
+        #  * Any subscriber that would show up in `buddy_data.get_filtered_user_id_list`,
+        #    which determines who is shown in the buddy list. This ensures we show them
+        #    in the correct section. We should be careful to keep this code in sync with
+        #    that buddy list logic.
+        #  * Subscriptions for all bots, so that we can calculate human subscriber
+        #    counts on the client.
         partial_subscriber_map: dict[int, list[int]] = dict()
+
+        # This constant should stay in sync with prescence.ts
+        # (send it from the server?)
+        BIG_REALM_COUNT = 250
+        active_users = list(user_profile.realm.get_active_users())
+        send_partial_subscribers = (
+            include_subscribers == "partial" and len(active_users) >= BIG_REALM_COUNT
+        )
+
+        def get_recently_active_user_ids() -> set[int]:
+            recent = timezone_now() - timedelta(seconds=settings.OFFLINE_THRESHOLD_SECS)
+            rows = UserPresence.objects.filter(
+                last_connected_time__gte=recent,
+            ).values("user_profile_id")
+            online_user_ids = {row["user_profile_id"] for row in rows}
+            return set(online_user_ids)
+
         if send_partial_subscribers:
+            # This constant should stay in sync with buddy_data.ts
+            # (send it from the server?)
+            MAX_CHANNEL_SIZE_TO_SHOW_ALL_SUBSCRIBERS = 75
             bot_users = set(
                 UserProfile.objects.filter(
                     is_bot=True, realm=user_profile.realm, is_active=True
                 ).values_list("id", flat=True)
             )
+            online_users = get_recently_active_user_ids()
```

Comment:
So then I think this query would be `UserProfile.objects.fliter(realm=realm, is_active=True, is_bot | not long_term_idle)` as pseudocode; that's not actually valid syntax; you need to use Django's `Q()` I think to do that union properly.

Reply:
resolving this in favor of your raw sql above